primitives/header-chain/src/lib.rs

```diff
@@ -190,13 +193,48 @@ impl<Number: Codec> ConsensusLogReader for GrandpaConsensusLogReader<Number> {
 	}
 }
 
-/// The Grandpa-related info associated to a header.
+/// The finality-related info associated to a header.
 #[derive(Encode, Decode, Debug, PartialEq, Clone, TypeInfo)]
-pub struct HeaderGrandpaInfo<Header: HeaderT> {
-	/// The header justification
-	pub justification: justification::GrandpaJustification<Header>,
-	/// The authority set introduced by the header.
-	pub authority_set: Option<AuthoritySet>,
+pub struct HeaderFinalityInfo<FinalityProof, FinalityVerificationContext> {
+	/// The header finality proof.
+	pub finality_proof: FinalityProof,
+	/// The new validation context introduced by the header.
+	pub new_verification_context: Option<FinalityVerificationContext>,
+}
+
+/// Grandpa-related info associated to a header. This info can be saved to storage.
```

Comment:
I think it can only be saved to events, not to the storage :) But I wonder if we can use the single `HeaderGrandpaInfo` type for both saving and verification? Is there some significant difference? Why can't we e.g. use `AuthoritySet` in verification and convert it to `VoterSet` when required?